bundles/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/discovery/RokuDiscoveryService.java

```diff
@@ -0,0 +1,267 @@
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("Error discovering devices", e);
+        }
+    }
+
+    /**
+     * Broadcasts a SSDP discovery message into the network to find provided services.
+     *
+     * @return The Socket where answers to the discovery broadcast arrive
+     */
+    private @Nullable DatagramSocket sendDiscoveryBroacast(NetworkInterface ni) {
+        try {
+            InetAddress m = InetAddress.getByName("239.255.255.250");
+            final int port = 1900;
+
+            if (!ni.isUp() || !ni.supportsMulticast()) {
+                return null;
+            }
+
+            Enumeration<InetAddress> addrs = ni.getInetAddresses();
+            InetAddress a = null;
+            while (addrs.hasMoreElements()) {
+                a = addrs.nextElement();
+                if (a instanceof Inet4Address) {
+                    break;
+                } else {
+                    a = null;
+                }
+            }
+            if (a == null) {
+                logger.debug("No ipv4 address on {}", ni.getName());
+                return null;
+            }
+
+            // Create the discovery message packet
+            byte[] requestMessage = ROKU_DISCOVERY_MESSAGE.getBytes(StandardCharsets.UTF_8);
+            DatagramPacket datagramPacket = new DatagramPacket(requestMessage, requestMessage.length, m, port);
+
+            // Create socket and send the discovery message
+            DatagramSocket socket = new DatagramSocket();
+            socket.setSoTimeout(3000);
+            socket.send(datagramPacket);
+            return socket;
+        } catch (IOException e) {
+            logger.debug("sendDiscoveryBroacast() got IOException: {}", e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and process those that come from a Roku.
+     *
+     * @param socket The socket where answers to the discovery broadcast arrive
+     */
+    private void scanResposesForKeywords(DatagramSocket socket) {
+        boolean socketTimeout = false;
+        do {
+            byte[] receiveData = new byte[1024];
+            DatagramPacket packet = new DatagramPacket(receiveData, receiveData.length);
+            try {
+                socket.receive(packet);
+            } catch (SocketTimeoutException e) {
+                socket.close();
+                socketTimeout = true;
+                return;
+            } catch (IOException e) {
+                logger.debug("Got exception while trying to receive UPnP packets: {}", e.getMessage());
+                socket.close();
+                socketTimeout = true;
+                return;
+            }
+            String response = new String(packet.getData(), StandardCharsets.UTF_8);
+            if (response.contains(ROKU_SSDP_MATCH)) {
+                parseResponseCreateThing(response);
+            }
+        } while (!socketTimeout);
```

Comment:
This seems a bit overkill. Because `socketTimeout` will never be true. Because if it's made true the method will call return. So you can remove this variable and simply make it a `while(true))` loop.